pallets/parachain-staking/src/lib.rs

```diff
@@ -1809,27 +1817,37 @@ pub mod pallet {
 			Ok(())
 		}
 
-		pub(crate) fn prepare_staking_payouts(now: RoundIndex) -> Weight {
-			// payout is now - delay rounds ago => now - delay > 0 else return early
-			let delay = T::RewardPaymentDelay::get();
-			if now <= delay {
-				return Weight::zero();
-			}
-			let round_to_payout = now.saturating_sub(delay);
-			let total_points = <Points<T>>::get(round_to_payout);
-			if total_points.is_zero() {
+		pub(crate) fn prepare_staking_payouts(
+			round_info: RoundInfo<BlockNumberFor<T>>,
+			round_duration: u64,
+		) -> Weight {
+			let RoundInfo {
+				current: now,
+				length: round_length,
+				..
+			} = round_info;
+
+			// This function is called right after the round index increment,
+			// and the goal is to compute the payout informations for the round that just ended.
+			// We don't need to saturate ehre because the genesis round is 1.
+			let prepare_payout_for_round = now - 1;
```

Comment:
Shouldn't this be set to `now - RewardPaymentDelay`? Why is the subtraction being forced to `1`?

Reply:
@Agusrodri this code is exeuted juts after the round index increment, so at the end of the round N, now equal N + 1, so we should fill the data related to the round N = now -1

Reply:
I see. So given that, should we keep managing the same logic inside `handle_delayed_payouts`? Because in that function we are still paying the rewards taking into account the `RewardPaymentDelay` constant. Is it okay if the process of preparing the payouts and the process of actually distributing the rewards are independent to each other in terms of delay?

Reply:
>  Is it okay if the process of preparing the payouts and the process of actually distributing the rewards are independent to each other in terms of delay?

I think so, I don't see any technical reason to link the two. The only technical requirement is that the payout informations for a round should be filled in before the rewards distribution